### PR TITLE
Expose download_pkgs rule's impl, attrs and outputs.

### PR DESCRIPTION
**Pull request by xingao267**

### CODE REVIEW

File: package_managers/download_pkgs.bzl [hunk 1]
```diff
@@ -31,28 +34,47 @@ apt-get install --no-install-recommends -y -q -o Dir::Cache="/tmp/install" -o Di
...
-        outputs = [ctx.outputs.pkg_tar],
-        executable = ctx.outputs.build_script,
-        inputs = [ctx.file.image_tar],
+        outputs = [output_tar],
+        executable = output_script,
+        inputs = [image_tar],
     )
 
-def _impl(ctx):
+def _impl(ctx, image_tar=None, packages=None, additional_repos=None, output_executable=None, output_tar=None, output_script=None):
```
Comment (tejal29) [inline package_managers/download_pkgs.bzl]: same here can we keep the _impl signature same and add another function.

File: package_managers/download_pkgs.bzl [hunk 2]
```diff
@@ -66,81 +88,60 @@ docker attach $cid
...
+    ),
+    "additional_repos": attr.string_list(),
+}
+
+_outputs = {
+    "pkg_tar": "%{name}.tar",
+    "build_script": "%{name}.sh",
+}
+
+download = struct(
```
Comment (tejal29) [inline package_managers/download_pkgs.bzl]: What does this struct do?

Comment (xingao267) [reply package_managers/download_pkgs.bzl]: This struct is to be loaded from another skylark rule. For example, in my rule, I do something like this:
load("@debian_docker//package_managers:download_pkgs.bzl", _download = "download")

Then I will have access to _download.attrs, _download.outputs and _download.implementation in my rule.

This is same pattern as what container_image rule is doing: https://github.com/bazelbuild/rules_docker/blob/master/container/image.bzl#L302
https://github.com/bazelbuild/rules_docker/blob/master/container/image.bzl#L21

Comment (tejal29) [reply package_managers/download_pkgs.bzl]: ah ok . Can you add a comment mentioning Exporting Download_pkgs rule for other bazel rules to depend on.

**Comment by tejal29** [comment]
@xingao267 Can you rebase from master and push to the branch. The error you are seeing in Kokoro might be fixed due to Dan's PR

File: package_managers/download_pkgs.bzl (hunk 2 above)
Comment (xingao267) [reply package_managers/download_pkgs.bzl]: Done.

**Comment by xingao267** [comment]
I rebased. But it still seems to fail. I tried it locally, and it was fine.

**Comment by tejal29** [comment]
Sorry, i just merged in @dlorenc  PR. He changed the /tmp/packages.tar hardcoded file target to name that has the rule name. 

Can you merge again?

Thanks
Tejal

**Comment by xingao267** [comment]
Yeah! PR #197 does fix it!

**Pull request merged**